trace_decoder/src/decoding.rs

```diff
@@ -582,8 +586,14 @@ fn process_txn_info(
         delta_out,
     )?;
 
+    let burn_addr = match use_burn_target {
+        // TODO: retrieve the actual burn address from cdk-erigon.
+        true => Some(H160::zero()),
+        false => None,
+    };
     let gen_inputs = GenerationInputs {
         txn_number_before: extra_data.txn_number_before,
+        burn_addr,
```

Comment:
I think we'd benefit from inversion of control here.

Right now we plumb a bool (and presumably data when the TODO is addressed), so that `trace_decoder` can change a field of `GenerationInputs`. 

But the caller has all in information they need (the burn address), and access to the place they should put it (GenerationInputs), so why don't we save the trouble and ask them to plumb the value through?

I.e instead of these (and future changes), do something like this:
```
let mut generation_inputs = entrypoint(..);
for gi in &mut generation_inputs {
    gi.burn_addr = ...
}
```

Reply:
I a a bit confused by this suggestion: to me, `process_txn_info` is the function in charge of providing correct `GenerationInputs`. The fact that the caller has the information beforehand (which I am not 100% sure will still be the case when we actually have to retrieve the information from the cdk-erigon chain... but I might well be wrong here), for me, does not change the fact that the function should return the correct values. This is especially the case since `entrypoint` is called in multiple places, and we might just forget to change the `GenerationInputs` one time.

Reply:
> I am not 100% sure will still be the case when we actually have to retrieve the information from the cdk-erigon chain... but I might well be wrong here

Does this mean that it will be `trace_decoder`'s responsibility to make an RPC call to an erigon node?

> This is especially the case since entrypoint is called in multiple places, and we might just forget to change the GenerationInputs one time.

This makes sense to me, let's keep your change as written for now

Reply:
> Does this mean that it will be trace_decoder's responsibility to make an RPC call to an erigon node?

The way I understand Linda's approach, once we actually can plug ourselves to some `cdk-erigon` node, we will have `zero-bin` RPC binary pulling the `burn_address` info, and then pass it (along all the other extra data necessary for the decoder to generate prover payloads, i.e. the `GenerationInputs`). Similarly to how we currently fetch + pass the info on the `checkpoint_state_trie_root` for instance.